src/scala/com/github/johnynek/bazel_deps/GeneratePom.scala

```diff
@@ -0,0 +1,108 @@
+package com.github.johnynek.bazel_deps
+
+import java.io.File
+
+import cats.implicits._
+import scala.concurrent.duration._
+import scala.util.{Failure, Success}
+
+object GeneratePom {
+  def readModel(path: File): Either[String, Model] = {
+    val content: Either[String, String] =
+      Model.readFile(path) match {
+        case Success(str) => Right(str)
+        case Failure(err) =>
+          Left(s"[ERROR]: Failed to read ${path}.\n$err")
+      }
+
+    content.right.flatMap { c =>
+      Decoders.decodeModel(Yaml, c) match {
+        case Right(m) => Right(m)
+        case Left(err) => Left(s"[ERROR]: Failed to parse ${path}.\n$err")
+      }
+    }
+  }
+
+  private def resolve[F[_]](model: Model, resolver: Resolver[F]): F[(Graph[MavenCoordinate, Unit])] = {
```

Comment:
mostly a copy and paste from [MakeDeps](https://github.com/johnynek/bazel-deps/blob/master/src/scala/com/github/johnynek/bazel_deps/MakeDeps.scala#L118)

Reply:
Can we share the code somehow instead? Can you make the part you are customizing a function that you accept?

Reply:
yeah, I think that's a great idea! I like your POC. Feel free to update it and I think we can merge that.

Reply:
Opened the POC as a [PR](https://github.com/johnynek/bazel-deps/pull/308) to be merged 😄 Thanks!